develop/CICD/Tutorials/CICD_Tutorial_Connector.md

```diff
@@ -0,0 +1,273 @@
+
+1. On the top, click on *dotnet.yml*. On the new window, click on the pencil icon to edit the page.
+
+1. Add a new Job called CD running on a windows image with 2 new steps that:
+
+    1. Only runs if the CI Job completed
+
+    1. Retrieves the source code
+
+    1. Installs the Packager and Deployer tools
+
+    1. Run the packager to create a .dmprotocol
+
+    1. Deploys the .dmprotocol to an accessible agent directly
+
+> [!IMPORTANT]
+> Deployment with a local artifact requires running on a windows OS. For more details you can take a look at this [github issue](https://github.com/SkylineCommunications/Skyline.DataMiner.CICD.Tools.DataMinerDeploy/issues/10)
+
+>[!IMPORTANT]
+> Deployment with a local artifact requires DataMiner 10.3.0/10.3.2 or higher
+
+```yml
+  CD:
+
+    runs-on: windows-latest
+    needs: CI
+    steps:
+    - uses: actions/checkout@v3
+  
+    - name: Install Package Creation
+      run: dotnet tool install -g Skyline.DataMiner.CICD.Tools.Packager
+    - name: Install DataMiner Deploy
+      run: dotnet tool install -g Skyline.DataMiner.CICD.Tools.DataMinerDeploy
+      
+    - name: Create Protocol Package
+      run: dataminer-package-create dmprotocol "${{ github.workspace }}" --name  "protocol" --output "${{ github.workspace }}\\_PackageResults"
+
+    - name: Direct Agent Deployment
+      run: dataminer-package-deploy from-artifact --path-to-artifact "${{ github.workspace }}\\_PackageResults\\protocol.dmprotocol" --dm-server-location "${{ secrets.SERVER_LOCATION }}" --dm-user "${{ secrets.DATAMINER_USER }}" --dm-password "${{ secrets.DATAMINER_PASSWORD }}"
+```
+
+1. Commit your changes
+
+1. Navigate to Actions and see the run of your enhanced pipeline. This will perform:
+
+    1. Compilation
+
+    1. Unit Testing
+
+    1. DataMiner Connector Validator
+
+    1. Package Creation
+
+    1. Direct Agent Deployment
+
+## Step 4: Add your GitHub secrets
+
+1. Navigate to Settings
+
+1. On the left, select *Secrets and variables*
+
+1. Under Actions, add a *New repository secret* for the following secrets:
+
+    1. SERVER_LOCATION
+
+    1. DATAMINER_USER
+
+    1. DATAMINER_PASSWORD
+
+## Step 5: Profit
+
+1. Navigate to Actions
+
+1. Select the *.NET* workflow on the left
+
+1. Click on *Run workflow*
+
+1. You should now see your CI and CD jobs complete successfully
+
+1. Taking a screenshot of this successful run and sending that to thunder@skyline.be or uploading it through the [DoJo tutorials page](https://community.dataminer.services/learning-courses-tutorials/) will grant you DevOps points.
+
+## Advanced Options: CI
+
+Skyline Communications organization uses more than these actions. For enhanced CI, You can take a look at our [reusable workflow](https://github.com/SkylineCommunications/_ReusableWorkflows/blob/main/.github/workflows/Connector%20Master%20SDK%20Workflow.yml):
+
+1. SonarCloud static analysis to our CI
+
+1. Ability to use our GitHub Organization private NuGet store
```

Comment:
This is only applicable when they have been added as external collaborator to the repository of that NuGet package.